src/main/java/io/bf2/kafka/authorizer/CustomAclAuthorizer.java

```diff
@@ -422,23 +422,23 @@ public List<CompletionStage<AclCreateResult>> createAcls(AuthorizableRequestCont
             .map(binding -> {
                 final CompletionStage<AclCreateResult> result;
 
-                if (!binding.entry().principal().startsWith(CustomAclBinding.USER_TYPE_PREFIX)) {
-                    log.info("Rejected attempt by user {} to create ACL binding with invalid principal name: {}",
+                String bindingPrincipal = binding.entry().principal();
+                if (!bindingPrincipal.startsWith(CustomAclBinding.USER_TYPE_PREFIX)) {
+                    log.info("Rejected attempt by user {} to create ACL binding with incorrect prefixing: {}",
                             requestContext.principal().getName(),
-                            binding.entry().principal());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_NON_USER_PRINCIPAL);
-                }
-                else if (hasPrincipalBindings(binding.entry().principal())) {
-                    /* Reject ACL operations as invalid where the principal named in the ACL binding is a principal with configured custom ACLs */
-                    log.info("Rejected attempt by user {} to create ACL binding for principal {} with existing custom ACL configuration",
+                            bindingPrincipal);
+                    result = errorResult(AclCreateResult::new, INVALID_ACL_PRINCIPAL_NON_USER_PREFIXED);
+                } else if (hasPrincipalBindings(bindingPrincipal)) {
+                    /* Reject ACL operations as invalid where the principal named in the ACL binding is a restricted principal with configured custom ACLs */
+                    log.info("Rejected attempt by user {} to create ACL binding for restricted principal {}",
                             requestContext.principal().getName(),
-                            binding.entry().principal());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL_MANAGED);
+                            bindingPrincipal);
+                    result = errorResult(AclCreateResult::new, INVALID_ACL_PRINCIPAL_RESTRICTED);
```

Comment:
Should we include the rejected principal here?

Reply:
Have included the principal in the error message. As discussed in gchat this is not leaking information as we are only returning the principal the user has sent to us.